src/Symfony/Component/Security/Http/FirewallMapInterface.php

```diff
@@ -35,4 +35,11 @@
      * @return array of the format array(array(AuthenticationListener), ExceptionListener)
      */
     public function getListeners(Request $request);
+
+    /**
+     * Cleans up the internal state of the firewall map.
+     *
+     * @param Request $request
+     */
+    public function detachListeners(Request $request);
```

Comment:
Adding a new method to an interface is a BC break and not allowed by our BC promise.